### ISSUE TITLE
Error adding field - "Fields must have same type"

### ISSUE THREAD

**Issue by ScottB-Ronco** [hougaard/Simple-Object-Designer#226]
I am adding a field called RNC_Ronco_HM with field caption Ronco HM # with the field type of TEXT to the following tables and pages

- Table 36 Sales Header
- Table 114 Sales Cr.Memo Header (created through field transfer)
- Table 5107 Sales Header Archive (created through field transfer)
- Page 44 Sales Credit Memo
- Page 6630 Sales Return Order
- Page 134 Posted Sales Credit Memo
- Page 6627 Sales Return Order Archive

However on a Sales Order when I try to post warehouse shipment related to sales order or try to post and invoice the sales orders,  I am getting the error "The Following fields must have the same type" as show in below image.

![image](https://github.com/hougaard/Simple-Object-Designer/assets/87865315/b38e5b4a-0b3f-40f8-bcba-be695dca630a)

All fields on all tables and pages are set to have the same field type as TEXT and character limit of 50 as shown in the below image. I first created the field on table 36 and then used field transfer to add it to tables 114 & 5107 and then from there I added it all the proper pages.

![image](https://github.com/hougaard/Simple-Object-Designer/assets/87865315/ecf7ef50-84a0-459c-892f-b05cb1c41fea)

Why am I getting the error that fields must have the same type? I've added similar basic fields like this before without any errors or problems in the past.

Final note I am on SOD version 9.0.0.303

Thank you
Scott

**Comment by hougaard** [comment]
Sometimes BC will do field transfer all to it self if it sees two fields with the same field number. So if you have two **different** fields, with the **same field number** you'll get that error

If "Add-on Order" is not a field in your app, then it might be a field from another app with 70005 as number, I suggest you give your field a different number.

**Comment by ScottB-Ronco** [comment]
OK with your answer it clarified things, but still not sure why the field numbers is an issue.

The field Add-On Order is a field that I create on Sales Header in SOD with field # 70005. The new field RNC_Ronco_HM I created on same table has field # 70007 and I created a transfer to have this field added to sales order archive table but SOD automatically created it as field # 70005 on this table.

Now the Add-On Order field is not being transferred to Sales Order Archive, so why would it cause this issue?
